### PR TITLE
Search docs updates

### PR DESCRIPTION
**Pull request by bryevdv**
* Updates the CHANGELOG for release
* Adds samples for faceted search
* Updates READMEs
* Improves some lacking ref docs

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_filter_query_async.py
```
Comment (brjohnstmsft) [inline sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py]: Should this match the name of the file?

File: sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+key = os.getenv("AZURE_SEARCH_API_KEY")
+
+async def filter_query():
+    # [START facet_query_async]
+    from azure.search.aio import SearchIndexClient
+    from azure.search import SearchApiKeyCredential, SearchQuery
+
+    search_client = SearchIndexClient(service_endpoint, index_name, SearchApiKeyCredential(key))
+
+    query = SearchQuery(search_text="WiFi", facets=["Category"])
```
Comment (brjohnstmsft) [inline sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py]: nit: If the sample is supposed to show getting *only* facets, then typically the app would pass top=0.

File: sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py (hunk 1 above)
Comment (bryevdv) [reply sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py]: Thank you, fixed in https://github.com/Azure/azure-sdk-for-python/pull/10207/commits/81e88769a8b010142e68e55ba187e319e9b49947

File: sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py (hunk 2 above)
Comment (bryevdv) [reply sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py]: updated in https://github.com/Azure/azure-sdk-for-python/pull/10207/commits/81e88769a8b010142e68e55ba187e319e9b49947

**Pull request merged**